sky/backends/backend_utils.py

```diff
@@ -1056,18 +1057,19 @@ def query_head_ip_with_retries(cluster_yaml: str, max_attempts: int = 1) -> str:
 @timeline.event
 def get_node_ips(cluster_yaml: str,
                  expected_num_nodes: int,
-                 handle: Optional[backends.Backend.ResourceHandle] = None,
+                 handle: backends.Backend.ResourceHandle,
                  head_ip_max_attempts: int = 1,
                  worker_ip_max_attempts: int = 1) -> List[str]:
     """Returns the IPs of all nodes in the cluster."""
 
+    assert handle is not None
     # When ray up launches TPU VM Pod, Pod workers (except for the head)
     # won't be connected to Ray cluster. Thus "ray get-worker-ips"
     # won't work and we need to query the node IPs with gcloud as
     # implmented in _get_tpu_vm_pod_ips.
     ray_config = common_utils.read_yaml(cluster_yaml)
-    use_tpu_vm = ray_config['provider'].get('_has_tpus', False)
-    if use_tpu_vm:
+    is_tpu_pod = tpu_utils.is_tpu_vm_pod(handle.launched_resources)
+    if is_tpu_pod:
```

Comment:
IIRC, we want to keep the `handle` argument optional, right? Could you explain why we can't check if the provisioning instance is a TPU pod without looking at `handle.launched_resources`? It looks like we can do this check by looking at `ray_config`.